packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,254 @@
-  dungeon: Ganon
-  events:
-    GANON_TRIAL_SHADOW: "has_lens && can_use_light_arrows && (has_hover_boots || (can_hookshot && has_fire && can_use_bow)) && silver_rupees_ganon_shadow"
-  locations:
-    "MQ Ganon Castle Shadow Trial Bomb Flower Chest": "(can_hookshot || has_hover_boots) && (can_use_bow || (has_lens && has_hover_boots && (can_use_din || has_bombflowers)))"
-    "MQ Ganon Castle Shadow Trial Switch Chest": "can_use_bow && has_lens && (has_hover_boots || (can_hookshot && has_fire))"
-    "MQ Ganon Castle SR Shadow Front-Right": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot))"
-    "MQ Ganon Castle SR Shadow Middle": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot))"
-    "MQ Ganon Castle SR Shadow Back-Left": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot && has_fire))"
-    "MQ Ganon Castle SR Shadow Back-Center": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot && has_fire))"
-    "MQ Ganon Castle SR Shadow Front-Center": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot))"
-    "MQ Ganon Pot Shadow End 1": "has_lens && (has_hover_boots || (can_hookshot && has_fire && can_use_bow)) && silver_rupees_ganon_shadow"
-    "MQ Ganon Pot Shadow End 2": "has_lens && (has_hover_boots || (can_hookshot && has_fire && can_use_bow)) && silver_rupees_ganon_shadow"
-    "MQ Ganon Castle Wonder Item": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot)) && (can_use_bow || has_explosives)"
+    "MQ Ganon Castle Spirit Trial Back Right Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Back Left Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Front Left Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Gold Gauntlets Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+"Ganon Castle Spirit Trial Ending Room":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Spirit Trial Sun Room Sun Side": "true"
+  events:
+    GANON_TRIAL_SPIRIT: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Spirit End 1": "true"
+    "MQ Ganon Pot Spirit End 2": "true"
+"Ganon Castle Shadow Trial Entrance Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Middle Beamos Area 1": "event(GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE)"
+    "Ganon Castle Main Room": "true"
+    "Ganon Castle Shadow Trial Ending Side": "climb_anywhere"
+  events:
+    GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE: "can_use_bow || ((longshot_anywhere || (has_lens && (has_hover_boots || hookshot_anywhere))) && (can_use_din || has_bombflowers))"
+    GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE: "longshot_anywhere || (has_lens && hookshot_anywhere) || (has_lens && ((can_hookshot && event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE)) || has_hover_boots))"
+  locations:
+    "MQ Ganon Castle Shadow Trial Bomb Flower Chest": "event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE) && (can_hookshot || has_hover_boots)"
```

Comment:
Child will need climb anywhere with hover boots.

Reply:
I can't get YL to grab the ledge at all, even with Climb ANywhere. the platform isn't tall enough for link to grab before falling (unlike the entrance platform which goes down to the floor).

Reply:
Added is_adult